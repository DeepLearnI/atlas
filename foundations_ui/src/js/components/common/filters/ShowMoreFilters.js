import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ShowMoreFilters extends Component {
  constructor(props) {
    super(props);
    this.state = {
      filters: this.props.filters,
      bubblesHidden: this.props.bubblesHidden,
    };
  }

  render() {
    const { filters, bubblesHidden } = this.state;

    const filterBubbles = [];
    for (let i = 1; i <= bubblesHidden; i += 1) {
      const index = filters.length - i;
      filterBubbles.push(
        <div key={filters[index].column} className="bubble inline-block">
          <p className="font-bold">
            {filters[index].column}:<span> {filters[index].value}</span>
          </p>
          <button type="button" className="close-button" />
        </div>,
      );
    }

    return (
      <div className="show-more-filters-container elevation-1">
        {filterBubbles}
      </div>
    );
  }
}

ShowMoreFilters.propTypes = {
  filters: PropTypes.array,
  bubblesHidden: PropTypes.number,
};

ShowMoreFilters.defaultProps = {
  filters: [],
  bubblesHidden: 0,
};

export default ShowMoreFilters;
