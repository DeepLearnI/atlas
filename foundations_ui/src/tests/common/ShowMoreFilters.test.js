import React from 'react';
import ReactDOM from 'react-dom';
import ShowMoreFilters from '../../js/components/common/filters/ShowMoreFilters';
import { shallow, mount } from 'enzyme';
import configureTests from '../setupTests';

configureTests();

it('Shallow Renders ShowMoreFilters', () => {
  const wrapper = shallow(<ShowMoreFilters/>);
  expect(wrapper).toMatchSnapshot();
});
