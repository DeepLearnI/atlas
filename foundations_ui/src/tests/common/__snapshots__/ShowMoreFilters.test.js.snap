// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders ShowMoreFilters 1`] = `
<div
  className="show-more-filters-container elevation-1"
/>
`;
